examples/carplay-web-app/src/useCarplayAudio.ts

```diff
@@ -18,24 +19,24 @@ const useCarplayAudio = (
   microphonePort: MessagePort,
 ) => {
   const [mic, setMic] = useState<WebMicrophone | null>(null)
-  const [audioPlayers] = useState(new Map<string, PcmPlayer>())
+  const [audioPlayers] = useState(new Map<AudioPlayerKey, PcmPlayer>())
 
   const getAudioPlayer = useCallback(
     (audio: AudioData): PcmPlayer => {
       const { decodeType, audioType } = audio
       const format = decodeTypeMap[decodeType]
-      const audioKey = [format.frequency, format.channel, audioType].join('_')
+      const audioKey = createAudioPlayerKey(decodeType, audioType)
       let player = audioPlayers.get(audioKey)
       if (player) return player
       player = new PcmPlayer(format.frequency, format.channel)
       audioPlayers.set(audioKey, player)
       player.volume(defaultAudioVolume)
       player.start()
       worker.postMessage({
-        type: 'audioPlayer',
+        type: 'audioBuffer',
```

Comment:
@gozmanyoni I don't get any audio with this set as 'audioBuffer' if I revert to the previous commit and leave it as 'audioPlayer' then audio appears to be working fine again

Reply:
Introduced a bug - fixed in the last commit :)

Reply:
I managed to get it implemented into react-CarPlay today, all audio artefacts are now gone when navigating and interacting with the ui 🎉